### PR TITLE
276743 api jb

### PR DESCRIPTION
**Pull request by jbarnsley10**

### CODE REVIEW

File: app/repos/dogs.js [hunk 1]
```diff
@@ -113,14 +113,68 @@ const addImportedDog = async (dog, transaction) => {
...
+    },
+    {
+      model: sequelize.models.status,
+      as: 'status'
+    }]
+  })
+  // Workaround due to Sequelize bug when using 'raw: true'
+  // Multiple rows aren't returned from an array when using 'raw: true'
+  // so the temporary solution is to omit 'raw: true' and parse the JSON yourself
+  return JSON.parse(JSON.stringify(dog))
```
Comment (sfitz42) [inline app/repos/dogs.js:167]: You should be able to just return JSON.stringify(dog) here.

File: app/routes/dog.js [hunk 2]
```diff
@@ -1,27 +1,23 @@
...
   }
 },
 {
   method: 'POST',
   path: '/dog',
   handler: async (request, h) => {
-    const payload = JSON.parse(request.payload)
-    await addImportedDog(payload.dog)
+    if (!request.payload?.dog) {
+      return h.response('bad request').code(400)
```
Comment (sfitz42) [inline app/routes/dog.js:17]: No need for 'bad request' inside the response.

**Review by sfitz42 — APPROVED** [review]
LGTM - Just a couple of comments

File: app/repos/dogs.js (hunk 1 above)
Comment (jbarnsley10) [reply app/repos/dogs.js]: Removed stringify altogether

**Pull request merged**